packages/build-utils/test/unit.get-project-paths.test.ts

```diff
@@ -0,0 +1,75 @@
+import path from 'path';
+import { normalizePath } from '../src';
+import { getProjectPaths, ProjectPath } from '../src/get-project-paths';
+import { FixtureFilesystem } from './utils/fixture-filesystem';
+
+describe.each<{
+  fixturePath: string;
+  resultPaths: ProjectPath[];
+  skipPaths?: ProjectPath[];
+  readdirCalls: number;
+  hasPathCalls: number;
+}>([
+  {
+    fixturePath: '32-monorepo-highly-nested',
+    resultPaths: [],
+    readdirCalls: 2,
+    hasPathCalls: 4,
+  },
+  {
+    fixturePath: '33-hybrid-monorepo',
+    resultPaths: ['backend/app-three'],
+    readdirCalls: 2,
+    hasPathCalls: 3,
+    skipPaths: ['frontend'],
+  },
+  {
+    fixturePath: '34-monorepo-no-workspaces',
+    resultPaths: ['backend/app-three', 'frontend/app-one', 'frontend/app-two'],
+    readdirCalls: 3,
+    hasPathCalls: 6,
+  },
+  {
+    fixturePath: '32-monorepo-highly-nested',
+    resultPaths: [],
+    readdirCalls: 2,
+    hasPathCalls: 4,
+  },
+  {
+    fixturePath: '35-no-monorepo',
+    resultPaths: [],
+    readdirCalls: 1,
+    hasPathCalls: 1,
+  },
+  {
+    fixturePath: '36-monorepo-some-nested',
+    resultPaths: ['frontend/app-two'],
+    readdirCalls: 2,
+    hasPathCalls: 4,
+  },
```

Comment:
Let's add a test fixture that gets one level deep. All of these tests are 0 levels or 2 levels as far as I can see